### PR TITLE
Tic tac toe

### PR DESCRIPTION
**Pull request by Likey00**
I added a TicTacToe Cog which contains a method (!tictactoe) that allows the user to play tic tac toe against an unbeatable AI using the minimax algorithm

### CODE REVIEW

File: cogs/tictactoecog.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+            
+            #Return moves at the end
+            return moves
+
+
+        #Checks if there's a draw on the board
+        def check_draw():
+
+            #If no valid moves, true, otherwise false
+            if len(valid_moves()) == 0:
```
Comment (rosalogia) [inline cogs/tictactoecog.py]: You don't need an if statement here, you can just `return len(valid_moves()) == 0`

**Review by rosalogia — CHANGES_REQUESTED** [review]
I think the code is quality and well-commented, but I think every part of the functionality being a nested function kind of hurts readability and obfuscates the way that the actual command works. Maybe that's just a personal thing, we can talk about it, but as I see it this would be more readable and clean if the nested functions were top-level functions that took the board as a parameter. I think the actual command function being very clean and simple, just a few function calls, would be ideal. If you're confused as to what I mean, I can show you. It shouldn't be more than a quick refactor.

**Comment by Likey00** [comment]
Aight I can fix the if statement thing (just a mistake from fast code) and also yea I'll just move all the functions outside, I had it like that initially but later decided it might be better if it was self contained

**Comment by rosalogia** [comment]
It might have been better like this if the tic tac toe cog didn't have it's own file, but since it does, it's not particularly messy to keep all the functions separately defined 👍

**Comment by rosalogia** [comment]
Final request on my behalf: please use docstrings under function names to document how they work as opposed to single line comments above function names. Thank you!

**Review by rosalogia — APPROVED** [review]

**Pull request merged**